grpc-proto/src/test/java/ValuesTest.java

```diff
@@ -0,0 +1,323 @@
+    }
+
+    @Test
+    public void invalid() {
+      assertThatThrownBy(
+              () -> {
+                Values.float_(Values.NULL);
+              })
+          .isInstanceOf(IllegalArgumentException.class);
+    }
+  }
+
+  @Nested
+  class DoubleTest {
+
+    @Test
+    public void encodeDecode() {
+      double expected = Double.MAX_VALUE;
+      assertThat(Values.double_(Values.of(expected))).isEqualTo(expected);
+    }
+
+    @Test
+    public void invalid() {
+      assertThatThrownBy(
+              () -> {
+                Values.double_(Values.NULL);
+              })
+          .isInstanceOf(IllegalArgumentException.class);
+    }
+  }
+
+  @Nested
+  class BytesTest {
+
+    @Test
+    public void encodeDecodeBytes() {
+      byte[] expected = new byte[] {1, 2, 3};
+      assertThat(Values.bytes(Values.of(expected))).isEqualTo(expected);
+    }
+
+    @Test
+    public void encodeDecodeByteBuffer() {
+      ByteBuffer expected = ByteBuffer.wrap(new byte[] {1, 2, 3});
+      assertThat(Values.byteBuffer(Values.of(expected))).isEqualTo(expected);
+    }
+
+    @Test
+    public void invalidBytes() {
+      assertThatThrownBy(
+              () -> {
+                Values.bytes(Values.NULL);
+              })
+          .isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    public void invalidByteBuffer() {
+      assertThatThrownBy(
+              () -> {
+                Values.byteBuffer(Values.NULL);
+              })
+          .isInstanceOf(IllegalArgumentException.class);
+    }
+  }
+
+  @Nested
+  class StringTest {
+
+    @Test
+    public void encodeDecode() {
+      String expected = "abc";
+      assertThat(Values.string(Values.of(expected))).isEqualTo(expected);
+    }
+
+    @Test
+    public void invalid() {
+      assertThatThrownBy(
+              () -> {
+                Values.string(Values.NULL);
+              })
+          .isInstanceOf(IllegalArgumentException.class);
+    }
+  }
+
+  @Nested
+  class UUIDTest {
+
+    @Test
+    public void encodeDecode() {
+      UUID expected = UUID.randomUUID();
+      assertThat(Values.uuid(Values.of(expected))).isEqualTo(expected);
+    }
+
+    @Test
+    public void invalid() {
+      assertThatThrownBy(
+              () -> {
+                Values.uuid(Values.NULL);
+              })
+          .isInstanceOf(IllegalArgumentException.class);
+    }
+  }
+
+  @Nested
+  class InetTest {
+
+    @Test
+    public void encodeDecodeV4() throws UnknownHostException {
+      InetAddress expected = InetAddress.getByName("127.0.0.1");
+      assertThat(Values.inet(Values.of(expected))).isEqualTo(expected);
+    }
+
+    @Test
+    public void encodeDecodeV6() throws UnknownHostException {
+      InetAddress expected = InetAddress.getByName("2001:db8::8a2e:370:7334");
+      assertThat(Values.inet(Values.of(expected))).isEqualTo(expected);
+    }
+
+    @Test
+    public void invalid() {
```

Comment:
Should probably also test for lengths other than 4 or 16 bytes (maybe take legit Ipv4/v6 value, append one more byte).

Reply:
done